regression/pages/common/utils.py

```diff
@@ -0,0 +1,526 @@
+                new_row = row.split(',')
+                coupons.update({new_row[0]: new_row[1]})
+    return coupons
+
+
+def skip_cleanup_if_test_passed():
+    """
+    Skip cleanup when tests are passing
+    """
+    def decorator(test_function):
+        """
+        Decorator
+        Args:
+            test_function:
+        """
+        @functools.wraps(test_function)
+        def wrapper(self, *args, **kwargs):
+            """
+            Wrapper
+            Args:
+                self:
+                *args:
+                **kwargs:
+            """
+            test_function(self, *args, **kwargs)
+            self.full_cleanup = False
+        return wrapper
+    return decorator
+
+
+def fill_input_fields(page, css_selectors, values):
+    """
+    A helper function to fill multiple fields in a form
+    Args:
+        page:
+        css_selectors:
+        values:
+    """
+    for css_selector, value in izip(css_selectors, values):
+        page.q(css=css_selector).fill(value)
+
+
+def select_value_from_drop_down(page, drop_down_name, value):
+    """
+    A helper function to select value from drop down
+    Args:
+        page:
+        drop_down_name:
+        value:
+    """
+    page.wait_for_element_visibility(
+        'select[name=' + drop_down_name + ']', 'Drop down is visible')
+    page.q(
+        css='select[name=' + drop_down_name + '] option[value="{}"]'.format(
+            value)).click()
+    page.wait_for(
+        lambda: page.q(
+            css='select[name=' + drop_down_name + '] option[value="{}"]'.
+            format(value)).selected,
+        "Correct value is selected",
+        timeout=DEFAULT_TIMEOUT
+    )
+
+
+def select_values_from_drop_downs(page, drop_down_names, values):
+    """
+    A helper function to select values from multiple drop downs on a page
+    Args:
+        page:
+        drop_down_names:
+        values:
+    """
+    for drop_down_name, value in izip(
+            drop_down_names, values
+    ):  # pylint: disable=cell-var-from-loop
+        page.wait_for_element_visibility(
+            'select[name=' + drop_down_name + ']', 'Drop down is visible')
+        page.q(
+            css='select[name=' + drop_down_name + '] option[value="{}"]'.
+            format(value)).click()
+        page.wait_for(
+            lambda: page.q(
+                css='select[name=' + drop_down_name + '] option[value="{}"]'.
+                format(value)).selected,
+            "Correct value is selected",
+            timeout=DEFAULT_TIMEOUT
+        )
+
+
+def click_checkbox(page, css_selector):
+    """
+    A helper function to click check boxes
+    Args:
+       page:
+       css_selector:
+    """
+    # pylint: disable=unnecessary-lambda
+    page.wait_for_element_visibility(css_selector, 'wait for target checkbox')
+    page_elem = page.q(css=css_selector).results[0]
+    if not page_elem.is_selected():
+        page.browser.execute_script("arguments[0].click();", page_elem)
```

Comment:
I was getting a lot of flakiness with selenium clicks on checkboxes, 4 out of 10 times the clicks just didn't work